### ISSUE TITLE
Aurelia specific logic

### ISSUE THREAD

**Issue by dbrezoev** [krausest/js-framework-benchmark#2]
Can you please, explain why we need this `long tsAfter = isAurelia ? tsEventFire : tsEvent;` Thanks.

**Comment by krausest** [comment]
I want to measure the duration from the click to that point of time when the DOM has been updated. Please take a look at the attached picture:
<img width="1183" alt="aurelia2" src="https://cloud.githubusercontent.com/assets/3374055/14259535/9dd360b8-faa7-11e5-96c4-56a66b37a083.png">
On the left is the click event, then a pause of ~90 msecs, then the repaint happens. I want to measure the duration starting with the click event until the final repaint occurred which means ~110 msecs. Aurelia causes a first very short repaint after the click event and thus I must wait for the timer event and take the next repaint. Calculating it that way comes very close to the duration that is shown in the timeline.

This also means that this benchmark favors frameworks that perform an immediate dom update. Thus especially the tests that run very short must be taken with caution, especially for aurelia since the measured duration is not determined by the javascript duration, but by the idle time waiting for the timer.

Vidom is another framework that batches dom updates, but vidom has a much shorter pause and seems to be using RAF instead of a timer and it does not cause a repaint after the event handler. Here's a timeline screenshot for vidom:
<img width="1226" alt="vidom_animationframe" src="https://cloud.githubusercontent.com/assets/3374055/14259388/f1a712a8-faa6-11e5-8d06-4e7d10aa7d62.png">
 
Aurelia was the only framework that forced me to create such a hack (at least for Chrome 48).

So my question is: Why does the timer for aurelia fire with such a long delay? Why don't they use the next animation frame?

**Comment by dbrezoev** [comment]
Thank you very much for the explanation. Sometimes when I run aurelia tests, the app throws exception saying that the last paint event is NULL. And the events ordering causes this ussue. It happens to be "EventDispatch" -> "PaintEvent" - "TimerFire" without another paint event afterwards. This happens rarely, if I run it 12 times it could appear in any of them or passes without exception. Is it appropriate to catch this exception and just ignore it as a workaround ? Thanks in advance

**Comment by krausest** [comment]
Looking again at the code I don't think it is needed anymore, since I changed the logic to take always the last paint. This was due to Chrome 49 which changed rendering and had paint events for other frameworks that were not shown in the timeline (and is quite a bit slower than Chrome 48). 

But that doesn't really explain your exception. I've only seen such problems when the browser window was hidden or minimized. Ignoring the exception works but doesn't leave the good feeling of correctness. Is the browser window visible? What Chrome Version are you using?
